pages/first-principles.markdown

```diff
@@ -106,6 +116,8 @@ On the third project, the cycle continues. You're constantly falling behind.
 
 The best option, in my experience, is for the organization to just accept deadlines don't work. That doesn't mean projects should take forever; practice Agile or MVP or what not.
 
+:speech_balloon: Hmm...it seems like the benefits of this principle come from (1) trusting that engineers will choose to make long term investments in the codebase and (2) providing enough autonomy that they can make those choices. If enough space is set aside for this, I don't see a problem with deadlines. In particular, Agile is all about making sprint commitments and hitting your definition of done by the end of the sprint. If those count as deadlines, we had plenty of them at the company where deadlines were rare. (Aside: can we not mention that company by name?) I will agree, however, that deadlines must originate from the people closest to the implementation, not be imposed on them.
```

Comment:
True, I guess there's a fine but important distinction between a deadline and a sense of urgency.

I think the issue with saying "just get rid of deadlines, your developers will be more productive" is that a release can depend on giving other teams enough lead time to write documentation, deal with press, train sales and customer support, etc.  How do you do this without turning non-deadlines into defacto deadlines?  Does this principle only apply to a narrow set of products/features, or is it universally true (outside of a small set of exceptions, perhaps)?